### ISSUE TITLE
API documentation question/suggestion

### ISSUE THREAD

**Issue by nathanmarks** [OptimalBits/bull#309]
Originally I thought the undocumented methods were to be considered private -- however I see them referenced a lot when searching through other issues as if they are public.

Is the documentation just incomplete and these methods are safe to use? (for eg, `queue.getJobs()` which I am using directly to get `paused` jobs, or any of the other `getXXX` methods for that matter).

There's also undocumented events such as `waiting` which I see is emitted in `Job.create()`.

If these methods are intended to be documented but aren't due to lack of time etc -- have you considered automatically generating markdown API documentation via jsdoc comments?

Eg:
https://github.com/nathanmarks/cbioportal-api-client#api-reference 
https://github.com/nathanmarks/stylishly/blob/master/docs/api/styleManager.md

You can pretty much do whatever you want with the output format via custom templates or even separate files per module. I'm using this library here: https://github.com/jsdoc2md/jsdoc-to-markdown

**Comment by chuym** [comment]
Yeah I'm familiar with jsdoc and I find it pretty a nice way to keep documentation since it lives together with the code. All in for it.
